### PR TITLE
feat(ui) Retrieve last ingested timestamp and display on frontend

### PR DESCRIPTION
**Pull request by chriscollins3456**
This PR gets the timestamp for when a dataset entity was last ingested by checking all of its aspects for the most recent `lastObserved` in its `systemMetadata` where the `runId` exists and is not the default (`no-run-id-provided`) which means that ingestion is what last touched this aspect. 

Here's what it's looking like on the frontend:

**within 1 week of ingestion, green**
image

**on hover**
image

**hovering over the help icon**
image

**Orange when it's more than a week but less than a month**
image

**Red when it's more than a month since**
image

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/chart/mappers/ChartMapper.java [hunk 1]
```diff
@@ -62,6 +63,9 @@ public Chart apply(@Nonnull final EntityResponse entityResponse) {
         result.setUrn(entityResponse.getUrn().toString());
         result.setType(EntityType.CHART);
         EnvelopedAspectMap aspectMap = entityResponse.getAspects();
+        Long lastIngested = SystemMetadataUtils.getLastIngested(aspectMap);
```
Comment (jjoyce0510) [inline datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/chart/mappers/ChartMapper.java:66]: Nice!

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/common/mappers/util/SystemMetadataUtils.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  }
+
+  public static Long getLastIngested(@Nonnull EnvelopedAspectMap aspectMap) {
+    Long lastIngested = null;
+    for (String aspect : aspectMap.keySet()) {
+      if (aspectMap.get(aspect).hasSystemMetadata()) {
+        SystemMetadata systemMetadata = aspectMap.get(aspect).getSystemMetadata();
+        if (systemMetadata.hasRunId() && !systemMetadata.getRunId().equals(DEFAULT_RUN_ID) && systemMetadata.hasLastObserved()) {
+          Long lastObserved = systemMetadata.getLastObserved();
+          if (lastIngested == null || lastObserved > lastIngested) {
```
Comment (jjoyce0510) [inline datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/common/mappers/util/SystemMetadataUtils.java:22]: Nice!

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/container/mappers/ContainerMapper.java [hunk 3]
```diff
@@ -39,6 +40,8 @@ public static Container map(final EntityResponse entityResponse) {
     final Container result = new Container();
     final Urn entityUrn = entityResponse.getUrn();
     final EnvelopedAspectMap aspects = entityResponse.getAspects();
+    Long lastIngested = SystemMetadataUtils.getLastIngested(aspects);
```
Comment (jjoyce0510) [inline datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/container/mappers/ContainerMapper.java:43]: One thing to keep in mind -- There is a possibility that in the future we become smarter about which aspects we fetch here. At that time we'll need to make some decisions...

File: datahub-web-react/src/app/entity/shared/containers/profile/header/PlatformContent/PlatformContentContainer.tsx [hunk 4]
```diff
@@ -5,6 +5,20 @@ import { useEntityData } from '../../../../EntityContext';
...
+import { GenericEntityProperties } from '../../../../types';
+import EntityRegistry from '../../../../../EntityRegistry';
+import { EntityType } from '../../../../../../../types.generated';
+
+export function getDisplayedEntityType(
+    entityData: GenericEntityProperties | null,
+    entityRegistry: EntityRegistry,
+    entityType: EntityType,
+) {
+    const entityTypeCased =
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/shared/containers/profile/header/PlatformContent/PlatformContentContainer.tsx:17]: Nice refactor!

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/container/mappers/ContainerMapper.java (hunk 3 above)
Comment (chriscollins3456) [reply datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/container/mappers/ContainerMapper.java:43]: totally - always good to keep that in mind!

**Review by jjoyce0510 — APPROVED** [review]
Nice work, @chriscollins3456 !

**Comment by xiphl** [comment]
Sorry for reviving this merged PR, @chriscollins3456 , but can I confirm if the logic considers TimeSeriesAspects as well? that is, if i programmatically ONLY form a time series aspect containing datasetStats and include a runId and lastObserved in the MCP, the last synchronized timestamp will also be updated?

**Pull request merged**